frontend/src/scenes/saved-insights/savedInsightsLogic.ts

```diff
@@ -292,14 +285,7 @@ export const savedInsightsLogic = kea<savedInsightsLogicType>([
         [insightsModel.actionTypes.renameInsightSuccess]: ({ item }) => {
             actions.setInsight(item)
         },
-        [dashboardsModel.actionTypes.updateDashboardInsight]: ({ insight }) => {
-            const matchingInsightIndex = values.insights.results.findIndex((i) => i.id === insight.id)
-            if (matchingInsightIndex >= 0) {
-                actions.setInsight(insight)
-            } else {
-                actions.addInsight(insight)
-            }
-        },
+        [dashboardsModel.actionTypes.updateDashboardInsight]: () => actions.loadInsights(),
```

Comment:
it's interesting that going over the wire gives you the correct result but using the insight passed into this action doesn't...

I'd suggest the action is being called incorrectly rather than we should add an API call - but it might be the API call is fast enough that we don't need to avoid it

even if it's fine to drop the insight and load it over the network, if the wrong insight is being passed in here then the calling logic has the wrong data and that logic needs to be fixed... no?

Reply:
I already spent quite a while trying to figure out the logics and fix it there...

Do you think it's acceptable to merge this as kind of a hotfix, so that customers at least have not the current confusion (we have at least 2 tickets about that)?
And then make another PR to try a better fix.

Reply:
progress over perfection 😊 

this area is too complicated - I remember the confusion of dealing with it very well 😅 

i'm too far from the problem to know if the hotfix is better or worse - just wanted to call out the consequence of the change (without assuming you knew or had spotted or not :))